src/build/buildTask.ts

```diff
@@ -121,14 +126,36 @@ export class BuildTask {
         : vscode.TaskRevealKind.Silent;
 
     if (!cmakeCacheExists) {
-      let compilerArgs = (idfConf.readParameter(
-        "idf.cmakeCompilerArgs",
-        this.currentWorkspace
-      ) as Array<string>) || [
-        "-G=Ninja",
-        "-DPYTHON_DEPS_CHECKED=1",
-        "-DESP_PLATFORM=1",
-      ];
+      const espIdfVersion = await getEspIdfFromCMake(this.idfPathDir);
+      let defaultCompilerArgs;
+      if (espIdfVersion === "x.x") {
+        Logger.warn(
+          "Could not determine ESP-IDF version. Using default compiler arguments for the latest known version."
+        );
+        defaultCompilerArgs = [
+          "-G=Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      } else if (compareVersion(espIdfVersion, "4.4") >= 0) {
+        defaultCompilerArgs = [
+          "-G=Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      } else {
+        defaultCompilerArgs = [
+          "-G",
+          "Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      }
+      let compilerArgs =
+        (idfConf.readParameter(
+          "idf.cmakeCompilerArgs",
+          this.currentWorkspace
+        ) as Array<string>) || defaultCompilerArgs;
```

Comment:
This PR also doesn't take into account that `-B=` and `-S=` will still break the behavior also.

Reply:
Good catch! I've fixed this as well.